azure-iot-device/tests/iothub/aio/test_async_clients.py

```diff
@@ -374,6 +377,62 @@ async def test_called_with_method_name_returns_method_request_from_named_method_
         assert inbox_mock.get.call_count == 1
         assert received_request is received_request
 
+    @pytest.mark.it("Locks the client to API Receive Mode if the receive mode has not yet been set")
+    @pytest.mark.parametrize(
+        "method_name",
+        [pytest.param(None, id="Generic Method"), pytest.param("method_x", id="Named Method")],
+    )
+    async def test_receive_mode_not_set(self, mocker, client, method_name):
+        # patch this so receive_method_request won't block
+        mocker.patch.object(
+            AsyncClientInbox, "get", return_value=(await create_completed_future(None))
```

Comment:
>create_completed_future [](start = 57, length = 23)

this is a mind-bender.  having create_completed_future be a coroutine seems silly -- I would expect `await create_completed_future()` to await on the completed future.
No action requested.  Just pointing out my confusion.

Reply:
I agree - fundamentally we have this problem in a lot of async tests - mocks just don't play well with coroutines

---
In reply to: [454526462](https://github.com/Azure/azure-iot-sdk-python/pull/620#discussion_r454526462) [](ancestors = 454526462)